aws/src/main/java/org/apache/iceberg/aws/s3/S3URI.java

```diff
@@ -50,7 +52,21 @@
    * @param location fully qualified URI
    */
   S3URI(String location) {
+    this(location, ImmutableMap.of());
+  }
+
+  /**
+   * Creates a new S3URI in the form of scheme://bucket/key?query#fragment
```

Comment:
the javadoc here does not really explain the additional mapping information. I think  the javadoc can be removed if they are duplicated, or we explain it as "Create a new S3URI with additional information of bucket to access point mapping"